kiwmi/desktop/output.c

```diff
@@ -279,6 +280,19 @@ output_destroy_notify(struct wl_listener *listener, void *UNUSED(data))
 
     wl_list_remove(&output->events.destroy.listener_list);
 
+    int n_layers = sizeof(output->layers) / sizeof(output->layers[0]);
+    for (int i = 0; i < n_layers; i++) {
+        struct kiwmi_layer *layer;
+        struct kiwmi_layer *tmp;
+        wl_list_for_each_safe (layer, tmp, &output->layers[i], link) {
+            wl_list_remove(&layer->link);
+            layer->output = NULL;
+            wlr_layer_surface_v1_destroy(layer->layer_surface);
+        }
+
+        wl_list_remove(&output->layers[i]);
```

Comment:
The `wl_list_remove` was useless, the list should be empty at that point and the memory is about to be freed any way.

`wlr_layer_surface_v1_destroy` is necessary because it tell clients that the layer is gone, like what the protocol tells us to do. And it calls the layer surface destroyed event in the compositor itself.

Reply:
> The `wl_list_remove` was useless, the list should be empty at that point and the memory is about to be freed any way.

I was only talking about this; thank GitHub for showing the context when a single line is being reviewed 😉